packages/edit-site/src/components/global-styles/screen-revisions/get-revision-changes.js

```diff
@@ -0,0 +1,174 @@
+	'settings.color': __( 'color settings' ),
+	'settings.typography': __( 'typography settings' ),
+	'color.text': __( 'text colors' ),
+	'color.background': __( 'background colors' ),
+	'spacing.margin': __( 'margin styles' ),
+	'spacing.padding': __( 'padding styles' ),
+	'spacing.blockGap': __( 'block gap' ),
+	'settings.layout': __( 'layout settings' ),
+	'typography.fontStyle': __( 'font style' ),
+	'typography.fontSize': __( 'font size' ),
+	'typography.lineHeight': __( 'line height' ),
+	'typography.fontFamily': __( 'font family' ),
+	'typography.fontWeight': __( 'font weight' ),
+};
+
+/**
+ * Returns an array of translated strings describing high level global styles and settings.
+ *
+ * @param {Object} revision
+ * @param {Object} revision.settings Global styles settings.
+ * @param {Object} revision.styles   Global styles.
+ * @return {string[] | []} An array of translated labels.
+ */
+export function getGlobalStylesChanges( { settings = {}, styles = {} } ) {
+	const changes = [];
+	if ( Object.keys( settings ).length > 0 ) {
+		changes.push( __( 'Global settings' ) );
+	}
+	Object.keys( styles ).forEach( ( key ) => {
+		if ( translationMap[ key ] ) {
+			changes.push( translationMap[ key ] );
+		}
+	} );
+
+	return changes;
+}
+
+function getTranslation( key, blockNames ) {
+	if ( translationMap[ key ] ) {
+		return translationMap[ key ];
+	}
+	const keyArray = key.split( '.' );
+
+	if ( keyArray?.[ 0 ] === 'blocks' ) {
+		const blockName = blockNames[ keyArray[ 1 ] ];
+		return sprintf(
+			// translators: %s: block name.
+			__( '%s block' ),
+			blockName
+		);
+	}
+
+	if ( keyArray?.[ 0 ] === 'elements' ) {
+		return sprintf(
+			// translators: %s: element name, e.g., heading button, link, caption.
+			__( '%s element' ),
+			translationMap[ keyArray[ 1 ] ]
+		);
+	}
+}
+
+function shuffle( array ) {
+	for ( let i = array.length - 1; i > 0; i-- ) {
+		// eslint-disable-next-line no-restricted-syntax
+		const j = Math.floor( Math.random() * ( i + 1 ) );
+		[ array[ i ], array[ j ] ] = [ array[ j ], array[ i ] ];
+	}
+	return array;
+}
+
+const cache = new Map();
+
+export function getRevisionChanges(
+	revision,
+	previousRevision,
+	blockNames,
+	maxResults = 5
+) {
+	if ( cache.has( revision.id ) ) {
+		return cache.get( revision.id );
+	}
+
+	const changedValueTree = deepCompare(
+		{
+			blocks: revision?.styles?.blocks,
+			color: revision?.styles?.color,
+			typography: revision?.styles?.typography,
+			spacing: revision?.styles?.spacing,
+			elements: revision?.styles?.elements,
+			settings: revision?.settings,
+		},
+		{
+			blocks: previousRevision?.styles?.blocks,
+			color: previousRevision?.styles?.color,
+			typography: previousRevision?.styles?.typography,
+			spacing: previousRevision?.styles?.spacing,
+			elements: previousRevision?.styles?.elements,
+			settings: previousRevision?.settings,
+		}
+	);
+
+	// Remove dupes and shuffle results.
+	const result = shuffle( [ ...new Set( changedValueTree ) ] )
```

Comment:
Why do results get shuffled? Apologies, I think this might have been mentioned elsewhere!

Reply:
Very good question. 

It's a bit of an artefact now that we've reduced the depth of reported changes. I left it in there in case I needed it again. 

I mainly put it there to ensure we saw a mix of changes from a large or repetitive change set. The comparison function starts at the top and works to the bottom. 

Before, even at three levels deep, the change set could have contained `n` x font size settings, or `n` x color palette changes. Under styles, the same styles could have changed over `n` revisions.

So shuffling the order prevented us rendering either just settings changes (because they're always at the start of the array due to their location in the global styles collection) or the same thing over and over again.

I think we can get rid of it, and make a call as to whether we need to reinstate to spice up the results.